cmake/Helpers.cmake

```diff
@@ -0,0 +1,685 @@
+FUNCTION(SET_INTERNAL VAR_NAME VALUE)
+    SET(${VAR_NAME} ${VALUE} CACHE INTERNAL "")
+ENDFUNCTION()
+
+FUNCTION(ADD_FLAG_IF_SUPPORTED)
+    SET(options "")
+    SET(oneValueArgs FLAG TEST_VALUE VALUE)
+    SET(multiValueArgs "")
+    CMAKE_PARSE_ARGUMENTS("" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    IF(${_TEST_VALUE})
+        SET(TEST_FLAG "${_FLAG}=${_TEST_VALUE}")
+    ELSEIF(${_VALUE})
+        SET(TEST_FLAG "${_FLAG}=${_VALUE}")
+    ELSE ()
+        SET(TEST_FLAG "${_FLAG}")
+    ENDIF ()
+    
+    STRING(TOUPPER "${TEST_FLAG}" FLAG_NAME)
+    CHECK_CXX_COMPILER_FLAG("${TEST_FLAG}" "${FLAG_NAME}")
+
+    IF("${FLAG_NAME}")
+        IF(${_VALUE})
+            SET(_FLAG "${_FLAG}=${_VALUE}")
+        ENDIF ()
+        STRING(APPEND CMAKE_CXX_FLAGS " ${_FLAG}")
+    ENDIF ()
+ENDFUNCTION()
+
+FUNCTION(USE_SCCACHE_IF_AVAILABLE)
+    FIND_PROGRAM(SCCACHE_PROGRAM "sccache")
+    IF(SCCACHE_PROGRAM)
+        MESSAGE(STATUS "Using sccache found at ${SCCACHE_PROGRAM} for caching build results")
+        SET(CMAKE_C_COMPILER_LAUNCHER "${SCCACHE_PROGRAM}" CACHE STRING "" FORCE)
+        SET(CMAKE_CXX_COMPILER_LAUNCHER "${SCCACHE_PROGRAM}" CACHE STRING "" FORCE)
+    ENDIF()
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_COMPILER)
+    INCLUDE(CheckCXXCompilerFlag)
+    IF(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+        IF((NOT DISABLE_WERROR) AND (NOT IS_RELEASE_VERSION))
+            STRING(JOIN " "
+                CMAKE_CXX_FLAGS
+                ${CMAKE_CXX_FLAGS}
+                "-Werror"
+                "-Wno-inline"
+                "-Wno-error=cast-align"
+                "-Wmissing-declarations"
+            )
+        ELSEIF(IS_RELEASE_VERSION)
+            ADD_FLAG_IF_SUPPORTED(
+                FLAG "-ffile-prefix-map"
+                TEST_VALUE "a=b"
```

Comment:
This is redundant, you can just use the final VALUE and then also slightly simplify ADD_FLAG_IF_SUPPORTED, since this is the only TEST_VALUE user.